packages/app/src/components/Dimensions/DataDimension/Item.js

```diff
@@ -0,0 +1,139 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { colors } from '../../../colors';
+import { RemoveSelectedItemButton } from './buttons';
+
+const styles = {
+    unselected: {
+        display: 'flex',
+        padding: '2px 5px 2px 2px',
+        borderRadius: 4,
+    },
+    selected: {
+        display: 'flex',
+        padding: '2px 5px 2px 2px',
+        backgroundColor: colors.lightBlue,
+        borderRadius: 4,
+    },
+    onHover: {
+        backgroundColor: '#92C9F7',
+    },
+    highlighted: {
+        backgroundColor: '#7EBFF5',
+        borderRadius: 4,
+    },
+    text: {
+        fontFamily: 'Roboto',
+        fontSize: 14,
+        padding: '0px 2px 1px 2px',
+    },
+    unselectedIcon: {
+        position: 'relative',
+        top: '30%',
+        left: '38%',
+        backgroundColor: colors.grey,
+        height: 6,
+        width: 6,
+    },
+    selectedIcon: {
+        position: 'relative',
+        top: '39%',
+        left: '40%',
+        backgroundColor: colors.accentPrimary,
+        height: 6,
+        width: 6,
+    },
+    iconContainer: {
+        minWidth: 20,
+    },
+};
+
+export const UnselectedIcon = () => {
+    return (
+        <div style={styles.iconContainer}>
+            <div style={styles.unselectedIcon} />
+        </div>
+    );
+};
+
+export const SelectedIcon = () => {
+    return (
+        <div style={styles.iconContainer}>
+            <div style={styles.selectedIcon} />
+        </div>
+    );
+};
+
+export class Item extends Component {
+    state = { mouseOver: false };
+
+    handleMouseEnter = () => {
+        this.setState({ mouseOver: true });
+    };
+
+    handleMouseLeave = () => {
+        this.setState({ mouseOver: false });
+    };
+
+    getItemStyle = () => {
+        let itemStyle = this.props.unselected
+            ? styles.unselected
+            : styles.selected;
+
+        if (this.state.mouseOver)
```

Comment:
Instead of doing the hover effect with javascript, use css. We already have the DataDimension.css file, you can add it there.